### PR TITLE
Make changes to update gatsby to latest

### PR DESCRIPTION
**Pull request by Bouncey**
I found a breaking change whilst trying to update gatsby, this PR separates Layout components from the `layouts` folder which enables an update. 

Refer [this comment](https://github.com/gatsbyjs/gatsby/issues/1978#issuecomment-328575300).

### CODE REVIEW

File: src/templates/Article.jsx [hunk 1]
```diff
@@ -10,13 +10,21 @@ const propTypes = {
...
+    frontmatter: {
+      title
+    }
+  } = article;
   return (
     <div>
       <Helmet>
         <title>{ `${title} | freeCodeCamp Guide` }</title>
         <link
-          href={ `https://freecodecamp.org/guide${props.location.pathname}` }
```
Comment (Bouncey) [inline src/templates/Article.jsx]: This failed on `build`

Comment (QuincyLarson) [reply src/templates/Article.jsx]: @Bouncey OK - I will wait to QA this until I get your go-ahead.

Comment (Bouncey) [reply src/templates/Article.jsx]: Oh, no it was just a note as to why it changed :smile: 

I have rebased this so it should be good to go

**Comment by Bouncey** [comment]
I found a breaking change whilst trying to update gatsby, this PR separates Layout components from the `layouts` folder which enables an update. 

Refer [this comment](https://github.com/gatsbyjs/gatsby/issues/1978#issuecomment-328575300).

@QuincyLarson @systimotic Could you please review this PR as it is holding up further work :+1:

**Comment by Bouncey** [comment]
The unexpected key warning is just due to a different build process. You would only see it again by installing `1.6.6` or below 👍

**Comment by QuincyLarson** [comment]
@Bouncey Awesome - thanks for figuring out the cause of this, and for updating to the latest version of Gatsby!

**Pull request merged**